org.testeditor.web.backend.persistence/src/test/java/org/testeditor/web/backend/persistence/TestExecutorTest.xtend

```diff
@@ -0,0 +1,129 @@
+package org.testeditor.web.backend.persistence
+
+import ch.qos.logback.classic.Level
+import ch.qos.logback.classic.Logger
+import ch.qos.logback.classic.spi.ILoggingEvent
+import ch.qos.logback.classic.spi.LoggingEvent
+import ch.qos.logback.core.Appender
+import java.util.regex.Pattern
+import org.assertj.core.api.Condition
+import org.junit.Assert
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.TemporaryFolder
+import org.mockito.ArgumentCaptor
+import org.mockito.Captor
+import org.mockito.InjectMocks
+import org.mockito.Mock
+import org.mockito.Mockito
+import org.mockito.MockitoAnnotations
+import org.slf4j.LoggerFactory
+import org.testeditor.web.backend.persistence.workspace.WorkspaceProvider
+
+import static org.assertj.core.api.Assertions.*
+import static org.mockito.Mockito.*
+
+class TestExecutorTest {
+
+	@Rule
+	public val temporaryFolder = new TemporaryFolder
+
+	@InjectMocks
+	TestExecutorProvider testExecutorProviderUnderTest
+
+	@Mock
+	WorkspaceProvider workspaceProviderMock
+	@Mock
+	Appender<ILoggingEvent> logAppender
+	@Captor
+	ArgumentCaptor<LoggingEvent> logCaptor
+
+	@Before
+	def void setupMocks() {
+		MockitoAnnotations.initMocks(this)
+		Mockito.when(workspaceProviderMock.workspace).thenReturn(temporaryFolder.root)
+	}
+
+	@Before
+	def void setupLogging() {
+		val logBackRootLogger = LoggerFactory.getLogger(Logger.ROOT_LOGGER_NAME) as Logger
+		logBackRootLogger.addAppender(logAppender)
+	}
+
+	@Test
+	def void testWrongFilenameFailsWithException() {
+		// given
+		val existingNonTestFile = 'actual-file.tl'
+		temporaryFolder.newFile(existingNonTestFile)
+
+		// when
+		try {
+			testExecutorProviderUnderTest.testExecutionBuilder('actual-file.tl')
+			Assert.fail('expected exception was NOT thrown!')
+		} catch (IllegalArgumentException exception) {
+			// then
+			assertThat(exception.message).contains(''''«existingNonTestFile»' is no test case'''.toString)
```

Comment:
I think using `ExpectectException` would be a bit more readable and also allows us to verify the message.